dipy/reconst/dti.py

```diff
@@ -824,12 +882,19 @@ def fit(self, data, mask=None):
             dti_params = params_in_mask.reshape(out_shape)
             if self.return_S0_hat:
                 S0_params = model_S0.reshape(out_shape[:-1])
+            if extra is not None:
+                for key in extra:
+                    self.extra[key] = extra[key].reshape(data.shape)
         else:
             dti_params = np.zeros(data.shape[:-1] + (12,))
             dti_params[mask, :] = params_in_mask
             if self.return_S0_hat:
                 S0_params = np.zeros(data.shape[:-1])
                 S0_params[mask] = model_S0.squeeze()
+            if extra is not None:
+                for key in extra:
+                    self.extra[key] = np.zeros(data.shape)
+                    self.extra[key][mask, :] = extra[key]
```

Comment:
Just to clarify: the assumption is that `extra` always returns as an array the shape of the data?

Reply:
I wonder if `extra` should be named something more specific.

Reply:
Thanks, @skoudoro had similar thoughts, and originally `extra` was only a single array, but I have now generalized to a dictionary whose entries must be the same shape as the data. The reason for this is that reshaping / flattening / masking that goes on.

I did consider to generalize further, but having made a fair bit of use of this extra functionality myself, I can't really see any use-cases where the extra information we want is not the same shape as the data. At least for now, this seems to be a good compromise that lays a template for future additions.

IMO, I think `extra` probably shouldn't be more specific, the entries could be quite different depending on the fitting function